x-pack/filebeat/module/aws/_meta/config.yml

```diff
@@ -33,4 +33,9 @@
     #var.queue_url: https://sqs.myregion.amazonaws.com/123456/myqueue
 
     # Profile name for aws credential
+    # Use "" to specify default
     #var.credential_profile_name: fb-aws
+
+    # filename of shared credential file
+    # Use "" to specify default
```

Comment:
Do we have to speicify "" to get default? If var.shared_credential_file is not specified in the aws.yml config, it should goes to default `~/.aws/credentials` right?

Reply:
@kaiyan-sheng  When I was testing if `var.shared_credential_file` wasn't set at all, I would get an error `template map has no entry for key`.   I tried testing for the existence of the key (if, with & index) but got the same error.  If we could conditionally test for the presence of the key that would be ideal.  Any ideas?

Reply:
In config template you can check if the variables is defined and handle it however you want.

```
shared_credential_file: {{ if .shared_credential_file }}{{ .shared_credential_file }}{{ end }}
```

This is my preferred way because you are not duplicating default values in several places.
```
{{ if .shared_credential_file }}
shared_credential_file: {{ .shared_credential_file }}
{{ end }}
```

```
shared_credential_file: {{ if .shared_credential_file }}{{ .shared_credential_file }}{{ else }}/some/file{{ end }}
```

Reply:
@andrewkroh It's likely I'm making a stupid mistake, but I tried the last one and got:

```
2020-01-20T20:47:59.256-0600	ERROR	instance/beat.go:921	Exiting: Error getting config for fileset aws/cloudtrail: Error interpreting the template of the input: template: text:6:6: executing "text" at <.shared_credential_file>: map has no entry for key "shared_credential_file"
Exiting: Error getting config for fileset aws/cloudtrail: Error interpreting the template of the input: template: text:6:6: executing "text" at <.shared_credential_file>: map has no entry for key "shared_credential_file"
```

I'm wondering if this is because of
```
	tpl := template.New("text").Option("missingkey=error")
```
line 263 in fileset.go . The missingkey=error got added in November.

Reply:
yep, changing to `missingkey=zero` made it so that 

```
{{ if .shared_credential_file }}
shared_credential_file: {{ .shared_credential_file }}
{{ end }}
```

worked.

Reply:
The variable isn't declared in the manifest. Try adding it to this file, but without a `default`. Then when its not set a zero value will be added to the template data. It should behave like [this](https://play.golang.org/p/EM5P2-5PJro).

https://github.com/elastic/beats/blob/9f986638e3c1d26a09f7ca3d6f52c5e6c713d04a/x-pack/filebeat/module/aws/cloudtrail/manifest.yml#L3-L5

Reply:
That worked.  Thanks.  Knew I was missing something.